src/encode/mod.rs

```diff
@@ -77,6 +79,20 @@ impl<'de> de::Deserialize<'de> for EncoderConfig {
 }
 
 /// A text or background color.
+#[cfg(feature = "file")]
+#[derive(Copy, Clone, Debug, Deserialize)]
+#[allow(missing_docs)]
+pub enum Color {
+    Black,
+    Red,
+    Green,
+    Yellow,
+    Blue,
+    Magenta,
+    Cyan,
+    White,
+}
+#[cfg(not(feature = "file"))]
```

Comment:
Seems like this block  is duplicated right?
maybe you're looking for something like `#[cfg_attr(feature = "file", derive(Deserialize))]`?

Reply:
Totally!  I didn't know about `#[cfg_attr(...)]`.